src/JSInterop/Microsoft.JSInterop/src/JSRuntimeBase.cs

```diff
@@ -88,9 +149,13 @@ internal void EndInvokeJS(long asyncHandle, bool succeeded, JSAsyncCallResult as
             {
                 if (!_pendingTasks.TryRemove(asyncHandle, out var tcs))
                 {
-                    throw new ArgumentException($"There is no pending task with handle '{asyncHandle}'.");
+                    // We should simply return if we can't find an id for the invocation.
+                    // This likely means that the method that initiated the call defined a timeout and stopped waiting.
+                    return;
                 }
 
+                CleanupRegistrations(asyncHandle);
```

Comment:
This is a bit confusing. It's referred to as `taskId` (not `asyncHandle`) everywhere else. Could this be updated to match?

Reply:
BTW I know that inconsistency was my fault.

Reply:
Sounds good

Reply:
Will do